crates/volta-core/src/tool/package/metadata.rs

```diff
@@ -260,6 +281,24 @@ impl BinConfig {
             read_to_string(&file).with_context(|| ErrorKind::ReadBinConfigError { file })?;
         RawBinConfig::from_json(config_src)?.try_into()
     }
+
+    pub fn from_file_if_exists(file: PathBuf) -> Fallible<Option<Self>> {
+        match read_to_string(&file) {
+            Err(error) => {
+                if error.kind() == io::ErrorKind::NotFound {
+                    Ok(None)
+                } else {
+                    Err(VoltaError::from_source(
+                        error,
+                        ErrorKind::ReadBinConfigError { file },
+                    ))
+                }
+            }
+            Ok(config_src) => RawBinConfig::from_json(config_src)?
+                .try_into()
+                .map(|config| Some(config)),
```

Comment:
Clippy raises a warning here: You actually don't need the closure `|config| Some(config)`, you can instead do this with:

```suggestion
                .map(Some),
```